src/app/admin/components/filter-search/filter-search.component.ts

```diff
@@ -0,0 +1,267 @@
+  @ViewChild('searchInput') searchInput: ElementRef;
+
+  filterMenu: string;
+  filterMenuDown = false;
+  collectionMenu: string;
+  collectionMenuDown = false;
+
+  constructor(
+    private collectionService: CollectionService,
+    private authService: AuthService,
+    private contextMenuService: ContextMenuService,
+    private toaster: ToasterService
+  ) {}
+
+  ngOnInit() {
+    this.getCollections();
+    this.findUserRestrictions();
+
+    // add the 'all' option into the list of statuses
+    this.statuses.splice(0, 0, 'All');
+
+    this.statuses = this.statuses.filter(
+      s => !['rejected', 'unreleased'].includes(s.toLowerCase())
+    );
+  }
+
+  /**
+   * Fetches the collections from the CollectionService and formats them for use in the context menu.
+   */
+  private getCollections(): void {
+    this.collectionService
+      .getCollections()
+      .then(collections => {
+        this.collections = collections;
+        this.collections.push({ abvName: 'all', name: 'All', hasLogo: false });
+
+        this.collections.sort((a, b) => {
+          if (a.name < b.name) {
+            return -1;
+          }
+          if (a.name > b.name) {
+            return 1;
+          }
+          return 0;
+        });
+      })
+      .catch(error => {
+        this.toaster.notify('Error!', error, 'bad', 'far fa-times');
+        console.error(error);
+      });
+  }
+
+  /**
+   * Checks for user's authorization
+   */
+  private findUserRestrictions() {
+    this.authService.user['accessGroups'].forEach((group: string) => {
+      if (group.includes('@')) {
+        this.isCollectionRestricted = true;
+      }
+    });
+  }
+
+  /**
+   * Set's the selected collection property to the full Collection object represented by the abbreviated name
+   * @param { string } abvName abbreviated name of the collection
+   */
+  setSelectedCollection(abvName: string) {
+    this._selectedCollection = this.collections.filter(
+      x => x.abvName === abvName
+    )[0];
+  }
+
+  /**
+   * Return the currently selected collection
+   *
+   * @readonly
+   * @type {Collection}
+   * @memberof FilterSearchComponent
+   */
+  get selectedCollection(): Collection {
+    return this._selectedCollection;
+  }
+
+  /**
+   * Hide or show the filter dropdown menu
+   * @param event {MouseEvent} the mouse event from clicking
+   */
+  toggleFilterMenu(event: MouseEvent) {
+    if (this.filterMenu) {
+      if (!this.filterMenuDown && event) {
+        this.contextMenuService.open(
+          this.filterMenu,
+          event.currentTarget as HTMLElement,
+          {
+            top: 10,
+            left:
+              (event.currentTarget as HTMLElement).getBoundingClientRect()
+                .width - 200
+          }
+        );
+      } else {
+        this.contextMenuService.destroy(this.filterMenu);
+      }
+
+      this.filterMenuDown = !!event;
+    } else {
+      console.error('Error! Attempted to use an unregistered context menu');
```

Comment:
Can you elaborate on this error? I see it's a console error, so it's meant for the developer. What action should they take to remedy this?

Reply:
The context menu component in its current state requires that any new context-menu's be registered with the context menu service before they can be used. Usually this occurs at the top of the file. I have a branch with this piece removed but I haven't finished it yet.